pilot/pkg/kube/inject/webhook.go

```diff
@@ -489,13 +489,19 @@ func (wh *Webhook) inject(ar *v1beta1.AdmissionReview) *v1beta1.AdmissionRespons
 		return toAdmissionResponse(err)
 	}
 
+	// Deal with potential empty fields, e.g., when the pod is created by a deployment
+	podName := potentialPodName(&pod.ObjectMeta)
+	if pod.ObjectMeta.Namespace == "" {
+		pod.ObjectMeta.Namespace = req.Namespace
```

Comment:
Do we want to mutate this? Maybe create a temp podNamespace and set that accordingly?

Reply:
@ostromart I thought about that too, the thing is that `ObjectMeta.Namespace` is also used within the other function, `injectRequired()`, so, I'd need to change it as well. This was the less intrusive change I found. I don't think it hurts, after all the pod namespace is the same as the req.

Anyway, let me know if you prefer not to change it so I can workaround it with more code changes.

Reply:
Ok, as long as the side effect is harmless it seems reasonable.